### PR TITLE
NIAD-1914: DocumentReference: No file is generated for placeholder attachments

### PR DESCRIPTION
**Pull request by bchotkowski**
Ticket -> [NIAD-1914](https://gpitbjss.atlassian.net/browse/NIAD-1914)

### CODE REVIEW

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java [hunk 1]
```diff
@@ -238,6 +237,9 @@ private void assertMultipleDocumentsRetrieved(String conversationId, int documen
         assertThat(documentList.size()).isEqualTo(documentCount);
     }
 
+    //TODO: Figure happy path for AbsentAttachments
```
Comment (rorysproule1) [inline e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java:240]: Do you need this comment left in?

**Review by rorysproule1 — APPROVED** [review]

**Review by rorysproule1 — APPROVED** [review]
Good job 😄

**Review by AlanSimpson1723 — APPROVED** [review]

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendAbsentAttachmentTaskDefinition.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import lombok.extern.jackson.Jacksonized;
+import uk.nhs.adaptors.gp2gp.common.task.TaskType;
+
+@Jacksonized
+@Getter
+@SuperBuilder
+@EqualsAndHashCode(callSuper = true)
+public class SendAbsentAttachmentTaskDefinition extends DocumentTaskDefinition {
+
+    public SendAbsentAttachmentTaskDefinition(DocumentTaskDefinitionBuilder<?, ?> b, String title) {
```
Comment (ElliotB-96) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendAbsentAttachmentTaskDefinition.java:17]: Pattern is generally to use builders on constructors, is this constructor neccessary?

Comment (bchotkowski) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendAbsentAttachmentTaskDefinition.java:17]: Found that weird too, but it was a resolution to InvalidDefinitionException being thrown

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendAbsentAttachmentTaskExecutor.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        storageConnectorService.uploadFile(storageDataWrapperWithMhsOutboundRequest, fileName);
+
+        var ehrExtractStatus = ehrExtractStatusService.updateEhrExtractStatusAccessDocument(
+            taskDefinition, fileName, taskId, messageId, getLengthOfBase64EncodedFileContent(fileContent)
+        );
+        detectTranslationCompleteService.beginSendingCompleteExtract(ehrExtractStatus);
+    }
+
+    private int getLengthOfBase64EncodedFileContent(String fileContent) {
+        return Base64.getEncoder().encode(fileContent.getBytes(StandardCharsets.UTF_8)).length;
```
Comment (ElliotB-96) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendAbsentAttachmentTaskExecutor.java:63]: There is now a util class containing a method for getting the length of bytes of the string, perhaps you can make use of that?

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AbsentAttachmentFileMapper.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        var absentAttachment = AbsentAttachmentParameters.builder()
+            .title(title)
+            .odsCode(odsCode)
+            .conversationId(conversationId)
+            .build();
+        return TemplateUtils.fillTemplate(ABSENT_ATTACHMENT_TEMPLATE, absentAttachment);
+    }
+
+
+
```
Comment (ElliotB-96) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AbsentAttachmentFileMapper.java:28]: Could you removed some of these empty lines

**Review by bbell-kainos — APPROVED** [review]

**Review by rorysproule1 — APPROVED** [review]

**Pull request merged**